CONTRIBUTING.md

```diff
@@ -1,13 +1,608 @@
+- [Contributor Cheat Sheet](https://git.k8s.io/community/contributors/guide/contributor-cheatsheet) - Common resources for existing developers
+
+## Mentorship
+
+- [Mentoring Initiatives](https://git.k8s.io/community/mentoring) - We have a diverse set of mentorship programs available that are always looking for volunteers!
+
+## Contact Information
+
+<!-- TODO(contact): Dedupe with README and k/community -->
+
+- [Slack channel](https://kubernetes.slack.com/messages/sig-release)
+- [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-sig-release)
+
+## Contributing Code
+
+To contribute bug fixes or features to SIG Release repositories:
+
+1. Communicate your intent.
+1. Make your changes.
+1. Test your changes.
+1. Update documentation and examples where appropriate.
+1. Open a Pull Request (PR).
+
+Communicating your intent lets SIG Release members know that you intend
+to contribute, and how. This sets you up for success - you can avoid duplicating
+an effort that may already be underway, adding a feature that may be rejected,
+or heading down a path that you would be steered away from at review time. The
+best way to communicate your intent is via a detailed GitHub issue. Take a look
+first to see if there's already an issue relating to the thing you'd like to
+contribute. If there isn't, please raise a new one! Let us know what you'd like
+to work on, and why. SIG Release members can't always triage new issues
+immediately, but we encourage you to bring them to our attention via one of our
+[communication channels][contact].
+
+> NOTE: Unlike the core [Kubernetes release cycle][kubernetes-release-cycle],
+> [SIG Release repositories][sig-release-subprojects] are versioned/released on
+> ad hoc (as needed basis), which means that the changes you've made will
+> not necessarily be immediately available for consumption in tagged artifacts.
+>
+> We may release components for the following, non-exhaustive reasons:
+>
+> - unblocking a core Kubernetes release cycle
+> - unblocking Kubernetes projects that leverage our libraries
+> - mitigating a security vulnerability
+> - critical bug fixes or regressions
+> - a feature is incomplete or does not work as intended
+>
+> If you need a new component release to unblock your project, please contact us to request one.
+
+<!--
+TODO(contribute): Review Crossplane release process for more inspiration
+ref: https://github.com/crossplane/crossplane/blob/master/docs/reference/release-cycle.md
+-->
+
+Be sure to practice [good git commit hygiene] as you make your changes. All but
+the smallest changes should be broken up into a few commits that tell a story.
+Use your git commits to provide context for the folks who will review PR, and
+the folks who will be spelunking the codebase in the months and years to come.
+Ensure each of your commits is signed-off by using `git commit -s`. SIG Release
```

Comment:
Out of curiosity, why do we "require" commits to be signed-off? This is usually a requirement for projects using DCO, but since we use CLA, does this matter at all?